osbs/core.py

```diff
@@ -630,24 +631,29 @@ def wait(self, build_id, states):
         #   1. our object was found and we are returning in the loop
         #   2. our object was not found and we keep waiting (in the loop)
         # Therefore, let's raise here
-        logger.error("build '%s' was not found during wait", build_id)
+        logger.warning("build '%s' was not found during wait", build_id)
         raise OsbsWatchBuildNotFound("build '%s' was not found and response stream ended" %
                                      build_id)
 
     def wait_for_build_to_finish(self, build_id):
-        for retry in range(1, 10):
+        for retry in range(144):
             try:
                 build_response = self.wait(build_id, BUILD_FINISHED_STATES)
                 return build_response
             except OsbsWatchBuildNotFound:
                 # this is woraround for https://github.com/openshift/origin/issues/2348
-                logger.error("I'm going to wait again. Retry #%d.", retry)
+                logger.warning("I'm going to wait again. Retry #%d.", retry)
                 continue
         raise OsbsException("Failed to wait for a build: %s" % build_id)
 
     def wait_for_build_to_get_scheduled(self, build_id):
-        build_response = self.wait(build_id, BUILD_FINISHED_STATES + BUILD_RUNNING_STATES)
-        return build_response
+        for retry in range(144):
+            try:
+                build_response = self.wait(build_id, BUILD_FINISHED_STATES + BUILD_RUNNING_STATES)
+                return build_response
+            except OsbsWatchBuildNotFound:
+                continue
+            raise OsbsException("Failed to schedule a build in 144 attempts: %s" % build_id)
```

Comment:
Worth putting 144 in a variable and using it both in the range() and in this exception message?

Reply:
Why 144 btw?

Reply:
watch resources will sleep for 30 seconds in each of 10 loops, so each invocation of watch_resource takes 5 minutes. looping 12 times can take an hour. looping 144 times (12 * 12) is 12 hours.

I guessed that most builds would be scheduled or finished within 12 hours.